cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/PartitionConsumerResult.java

```diff
@@ -24,9 +24,10 @@
  */
 public class PartitionConsumerResult {
   private final PartitionConsumerState partitionConsumerState;
-  private final Iterator<Partition> partitionIterator;
+  private final List<PartitionDetail> partitionIterator;
```

Comment:
what happens to partitions that expire? Are they dropped? Or are they returned again next time?